### PR TITLE
adjust use of braces to fix warnings with clang

### PR DESCRIPTION
**Pull request by wjwwood**
I was getting warnings about too many braces and too few. We'll see if this pull request introduces new warnings or otherwise breaks the Linux or Windows builds.

Connects to ros2/ros2#53

### CODE REVIEW

**Comment by dirk-thomas** [comment]
LGTM

Any chance that we will be able to run the Windows CI job on this soon?

**Comment by wjwwood** [comment]
> Any chance that we will be able to run the Windows CI job on this soon?

I'm going to look at it today, but I was pretty frustrated with it last I worked on it. My plan is to completely format the hard drive and start over...

File: test_communication/test/message_fixtures.hpp [hunk 1]
```diff
@@ -146,19 +146,19 @@ get_messages_dynamic_array_primitives()
...
-    msg->float64_values = {{1.11}};
-    msg->int8_values = {{127}};
-    msg->uint8_values = {{255}};
-    msg->int16_values = {{32767}};
-    msg->uint16_values = {{65535}};
-    msg->int32_values = {{2147483647}};
-    msg->uint32_values = {{4294967295}};
-    msg->int64_values = {{9223372036854775807}};
-    msg->uint64_values = {{18446744073709551615UL}};
+    msg->bool_values = {true};
```
Comment (esteve) [inline test_communication/test/message_fixtures.hpp]: Will this change work on Windows? IIRC the whole point of using double braces was to make Visual Studio happy.

Comment (wjwwood) [inline test_communication/test/message_fixtures.hpp]: Not sure, needs to be tested.

**Comment by tfoote** [comment]
+1 pending windows verification

**Comment by wjwwood** [comment]
Windows verification is blocked by the IDL error when run on Jenkins. I've run it manually and it gets past the IDL error, but fails with syntax errors on this package `test_communications`. So, I'll have to clear those errors before I can test this branch.

**Comment by wjwwood** [comment]
This appears to be working on Windows.

**Comment by esteve** [comment]
+1

**Pull request merged**